py_zipkin/zipkin.py

```diff
@@ -709,3 +713,125 @@ def create_http_headers_for_new_span(context_stack=None, tracer=None):
         'X-B3-Flags': '0',
         'X-B3-Sampled': '1' if zipkin_attrs.is_sampled else '0',
     }
+
+
+def is_valid_trace_id(candidate):
+    return candidate and re.match(r'(?:[0-9a-f]{16}|[0-9a-f]{32})$', candidate)
+
+
+def is_valid_span_id(candidate):
+    return candidate and re.match(r'[0-9a-f]{16}$', candidate)
+
+
+def _extract_zipkin_attrs_from_single_header(b3_header, sample_rate,
+                                             use_128bit_trace_id):
+    # b3={TraceId}-{SpanId}-{SamplingState}-{ParentSpanId}
+    # where the last two fields are optional.
+    bits = b3_header.split('-')
+    if len(bits) == 1:
+        # See if it's a valid encoding of (only) sample decision
+        if bits[0] == 'false' or bits[0] == '0':
+            sample_rate = 0.0
+        elif bits[0] == 'true' or bits[0] == '1':
+            sample_rate = 100.0
+        else:
+            # Nonsense gets a None
+            return None
+        return create_attrs_for_span(
+            sample_rate=sample_rate,
+            use_128bit_trace_id=use_128bit_trace_id)
+    if not is_valid_trace_id(bits[0]) or not is_valid_span_id(bits[1]):
+        return None
+    if len(bits) == 4:
+        if not is_valid_span_id(bits[3]):
+            return None
+        parent_span_id = bits[3]
+    else:
+        parent_span_id = None
+    if len(bits) >= 3 and bits[2]:
```

Comment:
add an empty line before this

Reply:
Done (including additional cleanup & commenting)